libs/novui/src/json-schema-components/templates/ArrayFieldTemplate.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {
+  ArrayFieldTitleProps,
+  ArrayFieldTemplateProps,
+  getTemplate,
+  getUiOptions,
+  ArrayFieldTemplateItemType,
+} from '@rjsf/utils';
+import { Grid } from '@mantine/core';
+import { css } from '../../../styled-system/css';
+import { Box, Flex, styled } from '../../../styled-system/jsx';
+import { title } from '../../../styled-system/recipes';
+
+const Title = styled('h2', title);
+
+export function ArrayFieldTemplate(props: ArrayFieldTemplateProps) {
+  const { canAdd, disabled, idSchema, uiSchema, items, onAddClick, readonly, registry, required, title, schema } =
+    props;
+  const {
+    ButtonTemplates: { AddButton },
+  } = registry.templates;
+  const uiOptions = getUiOptions(uiSchema);
+  const ArrayFieldTitleTemplate = getTemplate('ArrayFieldTitleTemplate', registry, uiOptions);
+  const ArrayFieldItemTemplate = getTemplate('ArrayFieldItemTemplate', registry, uiOptions);
+
+  return (
+    <Box>
+      <ArrayFieldTitleTemplate
+        idSchema={idSchema}
+        title={uiOptions.title || title}
+        schema={schema}
+        uiSchema={uiSchema}
+        required={required}
+        registry={registry}
+      />
+      {items.map(({ key, ...itemProps }) => {
+        return <ArrayFieldItemTemplate key={key} {...itemProps} />;
+      })}
+      {canAdd && <AddButton onClick={onAddClick} disabled={disabled || readonly} registry={registry} />}
+    </Box>
+  );
+}
+
+export function ArrayFieldTitleTemplate(props: ArrayFieldTitleProps) {
+  const { title } = props;
+
+  return (
+    <Title py={'75'} variant={'subsection'}>
+      {title}
+    </Title>
+  );
+}
+
+export function ArrayFieldItemTemplate(props: ArrayFieldTemplateItemType) {
+  const {
+    children,
+    disabled,
+    hasMoveDown,
+    hasMoveUp,
+    hasRemove,
+    index,
+    onDropIndexClick,
+    onReorderClick,
+    readonly,
+    registry,
+  } = props;
+  const { MoveDownButton, MoveUpButton, RemoveButton } = registry.templates.ButtonTemplates;
+
+  return (
+    <Grid key={index}>
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Is there an `id` or something more unique we can use for `key`? Using an `index` for a key can sometimes lead to frustrating, hard to diagnose bugs if re-ordering is possible‏